ros-rviz-image.html

```diff
@@ -0,0 +1,110 @@
+<link rel="import" href="../polymer/polymer.html">
+<link rel="import" href="../iron-image/iron-image.html">
+<link rel="import" href="../paper-checkbox/paper-checkbox.html">
+<link rel="import" href="../paper-input/paper-input.html">
+<link rel="import" href="imports.html">
+
+<dom-module id="ros-rviz-image">
+  <template>
+    <paper-input label="Image topic" value="{{topic}}"></paper-input>
+    <paper-checkbox checked="{{continuous}}">Continuously update</paper-checkbox>
+    <paper-input label="Transport hints" value="{{transport_hints}}"></paper-input>
+    <iron-image id="xy_image" src="[[image_source]]" alt="iron-image" ></iron-image>
+  </template>
+  <script>
+    Polymer({
+      is: 'ros-rviz-image',
+
+      properties: {
+        name: {
+          type: String,
+          value: 'Image',
+        },
+        topic: {
+          type: String,
+          value: '/camera/image/compressed',
+          notify: true,
+        },
+        continuous: {
+          type: Boolean,
+          value: true,
+          notify: true,
+        },
+        image_source: {
+          type: String,
+          value: '',
+        },
+        transport_hints: {
+          type: String,
+          value: 'compressed',
+          notify: true,
+        },
+        globalOptions: Object,
+        isShown: Boolean,
+        ros: Object,
+        tfClient: Object,
+        viewer: Object,
+      },
+
+      observers: [
+        '_optionsChanged(viewer, topic, continuous, transport_hints, ros)',
+      ],
+
+      destroy: function() {
+        // Nothing to destroy.
+      },
+
+      hide: function() {
+        if (this._imageTopic) {
+          this._imageTopic.unsubscribe();
+          this._imageTopic = null;
+        }
+      },
+
+      show: function() {
+        if (this.viewer && this.isShown) {
+          if (!this._imageTopic) {
+            this._updateDisplay();
+          }
+        }
+      },
+
+      _optionsChanged: function(viewer, topic, continuous, transport_hints, ros) {
+        this.hide();
+        this._updateDisplay();
+        this.show();
+      },
+
+      _updateDisplay: function(callback) {
+        console.log('Image updating display')
+        var that = this;
+
+        if (!(this.ros && this.viewer && this.topic &&
+             (this.continuous || this.continuous === false) &&
```

Comment:
I imagine this is taken from the occupancy grid, but I did some basic testing here and I think `(this.continuous || this.continuous === false)` will always be true. I'm not exactly sure what the `continuous` variable is supposed to be setting.